built_value_generator/lib/src/serializer_source_field.dart

```diff
@@ -154,9 +154,16 @@ abstract class SerializerSourceField
   }
 
   static String _generateCast(
-      String type, BuiltMap<String, String> classGenericBounds) {
+      String type, BuiltMap<String, String> classGenericBounds,
+      {bool topLevel: true}) {
     final bareType = _getBareType(type);
-    final generics = _getGenerics(type);
+
+    // For built collections we can cast to the bare type when deserializing,
+    // instead of the full generic type. This is because the `replace` method
+    // checks the generic type and copies if needed.
+    final generics = topLevel && DartTypes.isBuiltCollectionTypeName(bareType)
```

Comment:
Doesn't this add now lint warnings of BuiltMap<dynamic, dynamic>?

Reply:
Nope, no lint warnings. (the analyzer is run on presubmit/travis).

I guess it's allowed to use 'as' with dynamic, since it doesn't actually affect the type. (It remains BuiltMap<whatever>). The replace method accepts BuiltMap<dynamic, dynamic>, so everyone's happy.